From a6a59693066fd8da81f7107479df3e32a129247d Mon Sep 17 00:00:00 2001
From: Jean Delvare <jdelvare@suse.de>
Date: Wed, 6 Dec 2017 09:55:04 +0100
Subject: [PATCH] lib/Module.mk: Add missing dependencies

The lib symlinks lacked a dependency to the actual library file, so
parallel builds could run into a race and break.

Signed-off-by: Angelo Compagnucci <angelo@amarulasolutions.com>
---
 lib/Module.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/Module.mk b/lib/Module.mk
index 432a051..fd2c8c4 100644
--- a/lib/Module.mk
+++ b/lib/Module.mk
@@ -42,11 +42,11 @@ endif
 $(LIB_DIR)/$(LIB_SHLIBNAME): $(LIB_DIR)/smbus.o
 	$(CC) -shared $(LDFLAGS) -Wl,--version-script=$(LIB_DIR)/libi2c.map -Wl,-soname,$(LIB_SHSONAME) -o $@ $^ -lc
 
-$(LIB_DIR)/$(LIB_SHSONAME):
+$(LIB_DIR)/$(LIB_SHSONAME): $(LIB_DIR)/$(LIB_SHLIBNAME)
 	$(RM) $@
 	$(LN) $(LIB_SHLIBNAME) $@
 
-$(LIB_DIR)/$(LIB_SHBASENAME):
+$(LIB_DIR)/$(LIB_SHBASENAME): $(LIB_DIR)/$(LIB_SHLIBNAME)
 	$(RM) $@
 	$(LN) $(LIB_SHLIBNAME) $@
 
-- 
2.7.4

